### PR TITLE
Update RefreshIndicator documentation to clarify the need for the child to be scrollable

### LINKED ISSUE #22180: RefreshIndicator breaks when the ListView has a ScrollController
## Steps to Reproduce

```dart
import 'package:flutter/material.dart';

import 'dart:async';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Flutter Demo',
      theme: new ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: new TodosPage(),
    );
  }
}


class TodosPage extends StatefulWidget {
  TodosPage({Key key}) : super(key: key);
  @override
  _TodosPageState createState() => new _TodosPageState();
}

class _TodosPageState extends State<TodosPage> {
  final GlobalKey<RefreshIndicatorState> _refreshIndicatorKey =
      new GlobalKey<RefreshIndicatorState>();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: new AppBar(title: new Text("My Todos")),
      body: new RefreshIndicator(
        key: _refreshIndicatorKey,
          child: new ListView.builder(
            itemCount: 5, // Change to 50 and rebuild
            controller: ScrollController(), // Removing this makes pull to refresh work in both cases.
            itemBuilder: _itemBuilder,
          ),
        onRefresh: _onRefresh,
      ),
    );
  }

  // simulate a http request
  Future<Null> _onRefresh() {
    Completer<Null> completer = new Completer<Null>();

    new Timer(new Duration(seconds: 3), () {
      print("timer complete");
      completer.complete();
    });

    return completer.future;
  }

  Widget _itemBuilder(BuildContext context, int index) {
    return new ListTile(title: Text("Todo $index"));
  }
}

```
When using a ScrollController one ListView, OverscrollNotifications are not sent to RefreshIndicator when all list items fit on screen. As such you cannot pull to refresh.

Comments in the code indicate what you need to do to replicate this bug.

```
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.8.2 at /Users/tevjef/Desktop/Development/flutter
    • Framework revision 5ab9e70727 (2 weeks ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.2)
    • Android SDK at /usr/local/share/android-sdk
    • Android NDK at /usr/local/share/android-sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.2
    • ANDROID_HOME = /usr/local/share/android-sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.0

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.2 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.3 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.2.3)
    • IntelliJ at /Users/tevjef/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 28.0.4
    • Dart plugin version 182.4323.44

[!] VS Code (version 1.27.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • iPhone X • ED4365D3-5EA2-4556-8CDC-88E7A3C87FB1 • ios • iOS 11.4 (simulator)

! Doctor found issues in 1 category.

```

### PR DESCRIPTION
**Pull request by perclasson**
## Description

Update RefreshIndicator documentation to clarify the need for the child to be scrollable. Screenshot from documentation:

Screenshot 2020-02-10 at 10 47 50

## Related Issues

Closes https://github.com/flutter/flutter/issues/22180
## Tests

N/A, just updating documentation

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by rami-a — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 1]
```diff
@@ -56,18 +56,25 @@ enum _RefreshIndicatorMode {
...
 /// ListView(
 ///   physics: const AlwaysScrollableScrollPhysics(),
 ///   children: ...
-//  )
+/// )
 /// ```
 ///
-/// Using [AlwaysScrollableScrollPhysics] will ensure that the scroll view is
-/// always scrollable and, therefore, can trigger the [RefreshIndicator].
+/// A [ScrollView] is not scrollable by default if the [ScrollView.controller]
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/refresh_indicator.dart]: You might leave this comment out or reconsider it. The first point doesn't appear to be generally correct and the second point is maybe misleading - by default the refreshIndicator responds to its closest descendant per defaultScrollNotification and etc.

**Review by HansMuller — APPROVED** [review]
LGTM with some suggestions about the doc.

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 2]
```diff
@@ -56,18 +56,25 @@ enum _RefreshIndicatorMode {
 /// scrollable's contents and then complete the [Future] it returns. The refresh
 /// indicator disappears after the callback's [Future] has completed.
 ///
-/// If the [Scrollable] might not have enough content to overscroll, consider
-/// settings its `physics` property to [AlwaysScrollableScrollPhysics]:
+/// ## Troubleshooting
+///
+/// ### Refresh indicator does not show up
+///
+/// The [RefreshIndicator] requires the child to be scrollable. If the
```
[the remark this answers is not in the document]
Comment (perclasson) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: I think this suggestion makes sense, let me update the comment.

File: packages/flutter/lib/src/material/refresh_indicator.dart (hunk 1 above)
Comment (perclasson) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: I'll leave it this comment out. I think the above paragraph covers this now anyway.

**Pull request merged**